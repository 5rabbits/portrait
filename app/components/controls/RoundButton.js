import React, { PureComponent } from 'react';

export default class RoundButton extends PureComponent {
  render() {
    return (
      <a href="#" className="btn btn-outline-primary btn-sm block-rounded">
        <i className=`fa fa-fw fa-btn fa-${this.props.icon}`></i>
        { this.props.label }
      </a>
    );
  }
}
