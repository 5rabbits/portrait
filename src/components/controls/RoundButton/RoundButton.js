import React from 'react'
import PropTypes from 'prop-types'

const RoundButton = props =>
  <a href="#" className="btn btn-outline-primary btn-sm block-rounded">
    <i className={`fa fa-fw fa-btn fa-${props.icon}`} />
    {props.label}
  </a>

RoundButton.propTypes = {
  icon: PropTypes.string,
  label: PropTypes.string,
}

RoundButton.defaultProps = {
  icon: null,
  label: null,
}

export default RoundButton
