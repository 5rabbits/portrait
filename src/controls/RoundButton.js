import React from 'react'

const RoundButton = () =>
  <a href="#" className="btn btn-outline-primary btn-sm block-rounded">
    <i className={`fa fa-fw fa-btn fa-${this.props.icon}`} />
    {this.props.label}
  </a>

export default RoundButton
