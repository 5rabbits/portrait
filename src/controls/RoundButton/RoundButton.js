import React from 'react'

const RoundButton = props =>
  <a href="#" className="btn btn-outline-primary btn-sm block-rounded">
    <i className={`fa fa-fw fa-btn fa-${props.icon}`} />
    {props.label}
  </a>

export default RoundButton
