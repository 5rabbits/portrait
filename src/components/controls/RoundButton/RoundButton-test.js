import { React, toJSON, mount } from 'test/helper'
import RoundButton from 'controls/RoundButton'

describe('RoundButton', () => {
  it('should render with minimum props', () => {
    const component = mount(<RoundButton />)

    expect(toJSON(component)).toMatchSnapshot()
  })
})
