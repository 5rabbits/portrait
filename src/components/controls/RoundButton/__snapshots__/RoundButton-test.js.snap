// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton should render with minimum props 1`] = `
<RoundButton>
  <a
    className="btn btn-outline-primary btn-sm block-rounded"
    href="#"
  >
    <i
      className="fa fa-fw fa-btn fa-undefined"
    />
  </a>
</RoundButton>
`;
